npm/webpack-dev-server/src/helpers/angularHandler.ts

```diff
@@ -0,0 +1,237 @@
+import * as fs from 'fs-extra'
+import { tmpdir } from 'os'
+import * as path from 'path'
+import { pathToFileURL } from 'url'
+import type { PresetHandlerResult, WebpackDevServerConfig } from '../devServer'
+import { sourceDefaultWebpackDependencies } from './sourceRelativeWebpackModules'
+
+export type AngularJsonProjectConfig = {
+  projectType: string
+  root: string
+  sourceRoot: string
+  architect: {
+    build: {
+      options: { [key: string]: any } & { polyfills?: string }
+      configurations?: {
+        [configuration: string]: {
+          [key: string]: any
+        }
+      }
+    }
+  }
+}
+
+type AngularJson = {
+  defaultProject?: string
+  projects: {
+    [project: string]: AngularJsonProjectConfig
+  }
+}
+
+const dynamicImport = new Function('specifier', 'return import(specifier)')
+
+export async function angularHandler (devServerConfig: WebpackDevServerConfig): Promise<PresetHandlerResult> {
+  const webpackConfig = await getAngularCliWebpackConfig(devServerConfig)
+
+  return { frameworkConfig: webpackConfig, sourceWebpackModulesResult: sourceDefaultWebpackDependencies(devServerConfig) }
+}
+
+async function getAngularCliWebpackConfig (devServerConfig: WebpackDevServerConfig) {
+  const { projectRoot } = devServerConfig.cypressConfig
+
+  const {
+    generateBrowserWebpackConfigFromContext,
+    getCommonConfig,
+    getStylesConfig,
+  } = await getAngularCliModules(projectRoot)
+
+  const angularJson = await getAngularJson(projectRoot)
+
+  let { defaultProject } = angularJson
+
+  if (!defaultProject) {
+    defaultProject = Object.keys(angularJson.projects).find((name) => angularJson.projects[name].projectType === 'application')
+
+    if (!defaultProject) {
+      throw new Error('Could not find a project with projectType "application" in "angular.json"')
+    }
+  }
+
+  const defaultProjectConfig = angularJson.projects[defaultProject]
+
+  const tsConfig = await generateTsConfig(devServerConfig, defaultProjectConfig)
```

Comment:
We are generating a tmp tsConfig but another option would be to scaffold one to disk for them. There might be use cases for allowing the user to extend types or add TS files to the compilation which having one that the user owns would enable. Scaffolding it makes it part of our API, so if we want to make changes to it in the future it might be a breaking change. We could also provide an option for passing in a custom tsConfig.